### DISCUSSION TITLE
how to debug react query inside react native app?

### DISCUSSION THREAD

**Discussion by nikitph** [TanStack/query#4318] [Q&A]
Firstly great effort on the lib. its a very powerful tool. However i am curious how people debug the queries/mutations inside RN apps as the devtools dont work. A simple way to track the various queries being emitted and the relevant cache state is enough but i cant seem to figure out a way

**Comment by TkDodo** [comment]
the docs point to a 3rd party flipper plugin:
https://tanstack.com/query/v4/docs/react-native

**Reply by elevyg** [reply]
This worked for me in RQ v4 

```ts
import { QueryClient } from "@tanstack/react-query";
import { addPlugin } from "react-query-native-devtools";

 const [queryClient] = useState(() => {
    const client = new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          staleTime: Infinity,
          cacheTime: Infinity,
          networkMode: "online",
        },
      },
    });
    if (__DEV__) addPlugin({ queryClient: client });
    return client;
  });
```

**Comment by nikitph** [comment]
Thanks a lot. I will check it out.

**Comment by mefjuu** [comment]
Any ideas on debugging without Flipper? (removed as of 0.74 version of RN)
